src/t_list.c

```diff
@@ -394,28 +394,78 @@ void listElementsRemoved(client *c, robj *key, int where, robj *o) {
         dbDelete(c->db, key);
     }
     signalModifiedKey(c, c->db, key);
-    server.dirty++;
+    server.dirty += count;
 }
 
+/* Implements the generic list pop operation for LPOP/RPOP.
+ * The where argument specifies which end of the list is operated on. An
+ * optional count may be provided as the third argument of the client's
+ * command. */
 void popGenericCommand(client *c, int where) {
+    long count = 0;
+    robj *value;
+
+    /* Parse the optional count argument. */
+    if (c->argc == 3) {
+        if (getPositiveLongFromObjectOrReply(c,c->argv[2],&count,NULL) != C_OK) 
+            return;
+        if (count == 0) {
+            /* Fast exit path. */
+            addReplyNullArray(c);
+            return;
+        }
+    }
```

Comment:
We should throw a syntax error if you pass in extra arguments.

Reply:
Thanks! Corrected via a13da02. I was actually better with that stuff 3 years back :)